src/main/java/column/store/parquet/write/ParquetWriter.java

```diff
@@ -0,0 +1,140 @@
+import column.store.api.write.Writer;
+
+import com.globalmentor.apache.hadoop.fs.BareLocalFileSystem;
+
+public class ParquetWriter implements Writer  {
+
+  private final org.apache.parquet.hadoop.ParquetWriter<Object> writer;
+  private final BaseWriter[] columnWriters;
+
+  public ParquetWriter(final Path path, final Column... columns) throws IOException {
+    this.columnWriters = new BaseWriter[columns.length];
+    fillColumnWriters(columns);
+    this.writer = new BuilderImpl(path, columnWriters).build();
+  }
+
+  private void fillColumnWriters(final Column[] columns) {
+    for (int i = 0; i < columns.length; i++) {
+      var column = columns[i];
+      this.columnWriters[i] = switch (column) {
+        case BooleanColumn booleanColumn -> new BooleanWriter(booleanColumn, i);
+        case DoubleColumn doubleColumn -> new DoubleWriter(doubleColumn, i);
+        case IdColumn idColumn -> new IdWriter(idColumn, i);
+        case LongColumn longColumn -> new LongWriter(longColumn, i);
+        case StringColumn stringColumn -> new StringWriter(stringColumn, i);
+      };
+    }
+  }
+
+  @Override
+  public BooleanColumnWriter of(final BooleanColumn column) {
+    return Arrays.stream(columnWriters)
+            .filter(columnWriter -> columnWriter.column().equals(column))
+            .findFirst()
+            .map(columnWriter -> (BooleanColumnWriter) columnWriter)
+            .orElseThrow(() -> new NoSuchColumnException(column));
+  }
+
+  @Override
+  public DoubleColumnWriter of(final DoubleColumn column) {
+    return Arrays.stream(columnWriters)
+            .filter(columnWriter -> columnWriter.column().equals(column))
+            .findFirst()
+            .map(columnWriter -> (DoubleColumnWriter) columnWriter)
+            .orElseThrow(() -> new NoSuchColumnException(column));
+  }
+
+  @Override
+  public IdColumnWriter of(final IdColumn column) {
+    return Arrays.stream(columnWriters)
+            .filter(columnWriter -> columnWriter.column().equals(column))
+            .findFirst()
+            .map(columnWriter -> (IdColumnWriter) columnWriter)
+            .orElseThrow(() -> new NoSuchColumnException(column));
+  }
+
+  @Override
+  public LongColumnWriter of(final LongColumn column) {
+    return Arrays.stream(columnWriters)
+            .filter(columnWriter -> columnWriter.column().equals(column))
+            .findFirst()
+            .map(columnWriter -> (LongColumnWriter) columnWriter)
+            .orElseThrow(() -> new NoSuchColumnException(column));
+  }
+
+  @Override
+  public StringColumnWriter of(final StringColumn column) {
+    return Arrays.stream(columnWriters)
+            .filter(columnWriter -> columnWriter.column().equals(column))
+            .findFirst()
+            .map(columnWriter -> (StringColumnWriter) columnWriter)
+            .orElseThrow(() -> new NoSuchColumnException(column));
+  }
+
+  @Override
+  public void next() throws IOException {
```

Comment:
It would be possible, but `ParquetWriter` is supposed to be a shallow abstraction of parquet - if parquet does some buffering on its own, it will do it - if not, it's written immediately.